apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -21,13 +25,30 @@ export function useViewerI18n(locale: string) {
   );
 }
 
+function useClientLocale() {
+  const session = useSession();
+  // If the user is logged in, use their locale
+  if (session.data?.user.locale) return session.data.user.locale;
+  // If the user is not logged in, use the browser locale
+  if (typeof window !== "undefined") {
+    const browserLocale = parser.pick(locales, window.navigator.language, { loose: true });
```

Comment:
Why do we need to parse it client side? We are already handling parser pick on server side. Maybe I am missing something?

Reply:
To prevent duplicate caches. Without this we were generating caches for es, es-ES, es-MX, etc. when they all respond with the same data. Third ensures the initial request is always sent as 'es'. Also without this locales didn't load in time causing tests to fail due to revalidation.